### ISSUE TITLE
Specify IMDSv2 explicitly 

### ISSUE THREAD

**Issue by rlandesman** [runs-on/runs-on#22]
First of all, fantastic project thank you for putting this together.

I am running into an issue where a Service Control Policy of mine at the AWS Root level is blocking usage of IMDSv1 (as it is non-secure), however this is preventing ec2 instances from launching. Can we include an option (or just hard-coded) that newly launched instances use IMDSv2? Documentation: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/ec2/command/RunInstancesCommand/

Example SCP snippet:
```
    {
      "Effect": "Deny",
      "Action": "ec2:RunInstances",
      "Resource": "arn:aws:ec2:*:*:instance/*",
      "Condition": {
        "StringNotEquals": {
          "ec2:MetadataHttpTokens": "required"
        }
      }
    },
    {
      "Effect": "Deny",
      "Action": "ec2:RunInstances",
      "Resource": "arn:aws:ec2:*:*:instance/*",
      "Condition": {
        "NumericGreaterThan": {
          "ec2:MetadataHttpPutResponseHopLimit": "3"
        }
      }
    },
```

Error:

```
Error: Unable to start EC2 instance with the following configuration: {"imageSpec":{"owner":"135269210855","name":"runs-on-ubuntu22-full-x64-*","platform":"linux","arch":"x64"},"runnerSpec":{"cpu":2,"family":["m7a","m7g"]}}. The error was: UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:sts::<acct>:assumed-role/runs-on-RunsOnServiceRole-zNcO2LJFOLKr/45e75621-e0db-4f44-9ab3-ae5104eed917 is not authorized to perform: ec2:RunInstances on resource: arn:aws:ec2:us-east-1:<acct>:instance/* with an explicit deny in a service control policy. Encoded authorization failure message: ...
```

**Comment by rlandesman** [comment]
Thanks @crohr! As for the PR, wasn't feeling confident in the change and don't know how to test the code. Happy to re-open it if you think its good to go.

**Comment by crohr** [comment]
The change is available in v1.7.2, that just got released. Thanks!
